client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/steps/use-steps.tsx

```diff
@@ -0,0 +1,144 @@
+import { useTranslate } from 'i18n-calypso';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import type { Task, Expandable } from '@automattic/launchpad';
+
+const removeDuplicatedSlashes = ( url: string ) => url.replace( /(https?:\/\/)|(\/)+/g, '$1$2' );
+
+const getPluginInstallationPage = ( fromUrl: string ) => {
+	if ( fromUrl !== '' ) {
+		return removeDuplicatedSlashes(
+			`${ fromUrl }/wp-admin/plugin-install.php?s=%2522migrate%2520guru%2522&tab=search&type=term`
+		);
+	}
+
+	return 'https://wordpress.org/plugins/migrate-guru/';
+};
+
+const getMigrateGuruPageURL = ( siteURL: string ) =>
+	removeDuplicatedSlashes( `${ siteURL }/wp-admin/admin.php?page=migrateguru` );
+
+const recordInstructionsLinkClick = ( linkname: string ) => {
+	recordTracksEvent( 'calypso_site_migration_instructions_link_click', {
+		linkname,
+	} );
+};
+
+interface Options {
+	fromUrl: string;
+}
+
+interface Step {
+	task: Task;
+	expandable?: Expandable;
+}
+
+type Steps = Step[];
+
+export const useSteps = ( { fromUrl }: Options ): Steps => {
+	const translate = useTranslate();
+
+	const steps = [
+		{
+			title: translate( 'Install the Migrate Guru plugin' ),
+			content: (
+				<p>
+					{ translate(
+						"First you'll need to install and activate the {{a}}Migrate Guru plugin{{/a}} on the site you want to migrate. Click {{strong}}Next{{/strong}} when you're ready.",
+						{
+							components: {
+								strong: <strong />,
+								a: (
+									<ExternalLink
+										href={ getPluginInstallationPage( fromUrl ) }
+										target="_blank"
+										onClick={ () => recordInstructionsLinkClick( 'install-plugin' ) }
+									/>
+								),
+							},
+						}
+					) }
+				</p>
+			),
+		},
+		{
+			title: translate( 'Get your site ready' ),
+			content: (
+				<>
+					<p>
+						{ translate(
+							'Head to the {{a}}Migrate Guru plugin screen on your source site{{/a}}, enter your email address, and click {{strong}}%(migrateLabel)s{{/strong}}.',
+							{
+								components: {
+									strong: <strong />,
+									a: fromUrl ? (
+										<ExternalLink
+											href={ getMigrateGuruPageURL( fromUrl ) }
+											target="_blank"
+											onClick={ () => recordInstructionsLinkClick( 'go-to-plugin-page' ) }
+										/>
+									) : (
+										<strong />
+									),
+								},
+								args: { migrateLabel: 'Migrate' },
+							}
+						) }
+					</p>
+					<p>{ translate( 'Then, pick WordPress.com as your destination host.' ) }</p>
+					<p>
+						{ translate( 'All set? Click {{strong}}Next{{/strong}} below.', {
+							components: {
+								strong: <strong />,
+							},
+						} ) }
+					</p>
+				</>
+			),
+		},
+		{
+			title: translate( 'Add your migration key' ),
+			content: (
+				<>
+					<p>
+						{ translate(
+							'Copy the key below. Head to the Migrate Guru settings on your source site, and paste it into the {{strong}}Migration key{{/strong}} field.',
```

Comment:
Small nit, but the field is actually called _Migrate Guru Migration Key_ (looks like they're missing the space between _Migrate_ and _Guru_ 🤦🏻‍♀️ , but I think we should add the space in our's):

![Screenshot 2024-07-11 at 3 24 21 PM](https://github.com/Automattic/wp-calypso/assets/1190420/3b311838-f91a-4dfe-b8ce-d15212159664)

Reply:
ahah nice catch! I appreciate you following up through to the plugin to confirm the instructions are accurate ❤️ 

ahah nice catch! In this case do you think people will miss the field if we omit the MigrateGuru? I find it redundant to say Migrate Guru multiple times in a sentence.

Reply:
Probably not many, though I wouldn't be surprised if it tripped some people up, especially given that "Migration Key" falls at the end and not the beginning of the label.

Reply:
Good point. I agree that it won't be many but it's probably safer to include just in case. Thanks Donna!

Reply:
Thank you folks! I applied the suggested copy with the space. 😉
https://github.com/Automattic/wp-calypso/pull/92070/commits/e8dfd4c6a7f7373496624af99d28b98a962a2b2c

I also took the opportunity to extract this to an argument, so the field name is not translated, similar to the approach we're using for the "Migrate" button.